packages/docusaurus-theme-classic/src/index.ts

```diff
@@ -151,22 +154,38 @@ export default function docusaurusThemeClassic(
       return modules;
     },
 
-    configureWebpack() {
+    configureWebpack(config) {
       const prismLanguages = additionalLanguages
         .map((lang) => `prism-${lang}`)
         .join('|');
 
-      return {
-        plugins: [
-          // This allows better optimization by only bundling those components
-          // that the user actually needs, because the modules are dynamically
-          // required and can't be known during compile time.
-          new ContextReplacementPlugin(
-            /prismjs[\\/]components$/,
-            new RegExp(`^./(${prismLanguages})$`),
-          ),
-        ],
-      };
+      // This allows better optimization by only bundling those components
+      // that the user actually needs, because the modules are dynamically
+      // required and can't be known during compile time.
+      config.plugins?.push(
+        new ContextReplacementPlugin(
+          /prismjs[\\/]components$/,
+          new RegExp(`^./(${prismLanguages})$`),
+        ),
+      );
+
+      config.module?.rules?.forEach((rule) => {
```

Comment:
🤪 not a fan of this

Maybe we could just add an mdx-loader "admonitions" option on which we'd just forward the plugin options (if we keep them?)

I know it's not ideal to add everything to mdx-loader but it's probably a good-enough temporary solution until we figure out something better

Reply:
you can make the loader provide an optional "admonitions" feature that the theme would enable

it's less flexible but would definitively work without needing any extra lifecycle, giving us time to figure out a good API for that lifecycle. If we can avoid introducing temporary experimental public apis that's better

Reply:
> you can make the loader provide an optional "admonitions" feature that the theme would enable

Do you mean move the admonitions plugin to the mdx-loader or something else?